### DISCUSSION TITLE
[macOS 10.15 ] Installing Simulator with iOS 10

### DISCUSSION THREAD

**Discussion by igorgiumellizup** [actions/virtual-environments#4202] [Installed software]
Hello

I need to test my app in ci using Iphone 7 and OS 10.

I was checking the [installed Simulators](https://github.com/actions/virtual-environments/blob/main/images/macos/macos-10.15-Readme.md#installed-sdks) and noticed that there's no Simulator with iOS 10. Is it possible to download / create this Simulator? What do I need to download and config? Do I need to point to another Xcode version?

**Comment by dmitry-shibanov** [comment]
Hello @igorgiumellizup. You can install required Simulator or Xcode by [xcversion](https://github.com/xcpretty/xcode-install#simulators). Besides, you can take a look on this [issue](https://github.com/actions/virtual-environments/issues/551). I think after installation required simulators you also need to switch to the proper Xcode. You can do it by this snippet:
```
xcode-select --switch/Applications/Xcode_10.3.app/Contents/Developer
```

**Reply by igorgiumellizup** [reply]
Thanks, @dmitry-shibanov . But don't these operations require authenticating? For example ```xcversion simulators --install='iOS 10.3.1'```. At least [it says](https://github.com/xcpretty/xcode-install#simulators) it requires authentication

**Reply by dmitry-shibanov** [reply]
Hello @igorgiumellizup. Yes, you should define `XCODE_INSTALL_USER`, `XCODE_INSTALL_PASSWORD` as secret variables for your repository.

**Reply by igorgiumellizup** [reply]
@dmitry-shibanov, what are the values of these variables? Do I need to create an Apple dev / enterprise account of some sort or I just mention these vars with the "env" tag and git will use its built-in credentials?

**Reply by dmitry-shibanov** [reply]
Hello @igorFedotenkov. It should be credentials from Apple Developer account. You can set them up as [secret variables](https://docs.github.com/en/actions/security-guides/encrypted-secrets). After that you'll be able to use `xcversion`.

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
